packages/pigeon/lib/java_generator.dart

```diff
@@ -244,27 +263,35 @@ static MessageCodec<Object> getCodec() {
         indent.write('public void ${func.name}(Reply<$returnType> callback) ');
         sendArgument = 'null';
       } else {
-        final String argType = _javaTypeForDartType(func.arguments[0].type);
+        final Iterable<String> argTypes =
+            func.arguments.map((NamedType e) => _javaTypeForDartType(e.type));
+        final Iterable<String> argNames = indexMap(func.arguments,
+            (int index, NamedType arg) => _getArgumentName(index, arg) + 'Arg');
```

Comment:
This could also then share that function instead of duplicating the appending of `Arg`.

Reply:
Done.